### PR TITLE
feat: Slash provider task

### PR DESCRIPTION
**Pull request by snowmead**
Features:

- Has the blockchain service listen on the proofs dealer pallet `SlashableProvider` event.
- New `SlashProviderTask` slashes a single provider triggered by the SlashableProvider event bus.
- Adds `SlashProviderTask` to the BSP actor running tasks.
- Adds integration test for slashing providers.
- Adds `next_challenge_deadline` field in `NewChallengeCycleInitialised` for simpler processing on client side.
- `slash` extrinsic now accepts `provider_id: HashId<T>` instead of account id.

### CODE REVIEW

File: node/src/services/handler.rs [hunk 1]
```diff
@@ -128,6 +130,16 @@ where
             bsp_download_file_task.subscribe_to(&self.task_spawner, &self.file_transfer);
         remote_download_request_event_bus_listener.start();
 
+        // Slash your own kin or potentially commit seppuku on your own stake.
+        // Running this is as BSP is very honourable and shows a great sense of justice.
```
Comment (ffarall) [inline node/src/services/handler.rs:134]: 😂

File: test/suites/integration/bsp/slash-provider.test.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  });
+});
+
+/**
+ * Wait some time before sealing a block and checking if the provider was slashed.
+ * @param api
+ * @param providerId
+ */
+async function checkProviderWasSlashed(api: BspNetApi, providerId: string) {
+  // Weight for provider to be slashed.
```
Comment (ffarall) [inline test/suites/integration/bsp/slash-provider.test.ts:81]:
```suggestion
  // Wait for provider to be slashed.
```

**Review by ffarall — APPROVED** [review]
Great job! I appreciate the small corrections in comments and names in `proofs-dealer/src/utils.rs`

File: node/src/services/handler.rs [hunk 3]
```diff
@@ -128,6 +130,16 @@ where
             bsp_download_file_task.subscribe_to(&self.task_spawner, &self.file_transfer);
         remote_download_request_event_bus_listener.start();
 
+        // Slash your own kin or potentially commit seppuku on your own stake.
```
Comment (santikaplan) [inline node/src/services/handler.rs:133]: You either die as a samurai, or live long enough to become a villain 🙇‍♂️

**Pull request merged**